### PR TITLE
ktouch: init at 18.12.0

### PR DESCRIPTION
**Pull request by schmittlauch**
###### Motivation for this change

This is a WIP attempt to package the KTouch typing tutor.
A small fix for qtdeclarative was necessary to get it built (see respective commit), which causes plenty of rebuilds of dependants. 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by schmittlauch** [comment]
@ttuegel Do you know whether certain measures need to be taken when packaging QML apps like e.g. changing the QML2_IMPORT_PATH?
Currently this app fails at runtime not finding the kqtquickcharts QML module:

```
$ result/bin/ktouch
can't find resource: "schemata/data.xsd"
qCoreApp Application(0x7ffc8c92eac0)
qrc:/qml/main.qml:129:5: Type HomeScreen unavailable
qrc:/qml/homescreen/HomeScreen.qml:162:26: Type ProfileSelector unavailable
qrc:/qml/homescreen/ProfileSelector.qml:92:13: Type ProfileDetailsItem unavailable
qrc:/qml/homescreen/ProfileDetailsItem.qml:23:1: module "org.kde.charts" is not installed
```

**Comment by peterhoeg** [comment]
> Currently this app fails at runtime

Try installing the package using nix-env and run it from there.

**Comment by schmittlauch** [comment]
Just rebased this onto latest master. As long as I am not supposed to somehow fix these Qt/QML quirks of the package needing to be installed into a profile, I'd deem this **PR ready for review and merge**.

**Comment by schmittlauch** [comment]
I'd also create a backport to stable once this has been merged.

**Comment by schmittlauch** [comment]
Does @ttuegl still have some remarks to give, especially to the build fixes in qtdeclarative?
otherwise: ping @Mic92

**Comment by peterhoeg** [comment]
@GrahamcOfBorg test plasma5

**Comment by peterhoeg** [comment]
@grahamc, any idea about this grahamcofborg failure:

```
error: while evaluating 'recursiveUpdate' at /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/rbvermaa-spot/lib/attrsets.nix:424:26, called from /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/rbvermaa-spot/lib/attrsets.nix:157:28:
while evaluating 'recursiveUpdateUntil' at /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/rbvermaa-spot/lib/attrsets.nix:393:37, called from /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/rbvermaa-spot/lib/attrsets.nix:425:5:
while evaluating 'zipAttrsWith' at /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/rbvermaa-spot/lib/attrsets.nix:356:21, called from /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/rbvermaa-spot/lib/attrsets.nix:403:8:
while evaluating 'zipAttrsWithNames' at /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/rbvermaa-spot/lib/attrsets.nix:341:33, called from /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/rbvermaa-spot/lib/attrsets.nix:356:27:
while evaluating anonymous function at /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/rbvermaa-spot/lib/lists.nix:113:41, called from /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/rbvermaa-spot/lib/attrsets.nix:356:46:
while evaluating the attribute 'plasma5' at /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/rbvermaa-spot/nixos/tests/all-tests.nix:167:3:
while evaluating 'handleTest' at /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/rbvermaa-spot/nixos/tests/all-tests.nix:17:22, called from /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/rbvermaa-spot/nixos/tests/all-tests.nix:167:13:
while evaluating 'discoverTests' at /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/rbvermaa-spot/nixos/tests/all-tests.nix:13:19, called from /var/lib/gc-of-borg/.nix-test-rs/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/rbvermaa-spot/nixos/tests/all-tests.nix:18:5:
access to path '/nix/store/jqsr371wj4k8l9kisg70ic4gkgff0s5h-rbvermaa-spot' is forbidden in restricted mode
```

**Comment by Mic92** [comment]
This should work. `test` broke at some point:

@GrahamcOfBorg build nixosTests.plasma5

**Comment by Mic92** [comment]
Opened pull request to update the documentation: https://github.com/NixOS/ofborg/pull/300

**Comment by schmittlauch** [comment]
@ttuegel looks like I cannot change the merge target of an existing PR, so I'm going to open a new one: #53859

**Comment by schmittlauch** [comment]
@bjornfor @FRidh You mean my commits broke the build of xorg.oclock, or why are you bringing this up here?

**Comment by bjornfor** [comment]
@schmittlauch: I commented directly on a commit in master that didn't build. I guess the commit came from this PR, since github linked the comments...?

**Comment by schmittlauch** [comment]
TL;DR: This PR is closed, won't be merged and the commit you commented on didn't come from here.

@bjornfor No, not really. This PR was about packaging ktouch, and I suspect that @ttuegel somehow merged commits from staging into it, as this PR was supposed to go to staging instead of master.
But I saw that to late and closed this PR in favor of another one.
